### ISSUE TITLE
Poll: Who wants to join developing RP?

### ISSUE THREAD

**Issue by Christoph21x** [carlospuk/remotepotato#5]
Dear All, 

some of you forked the solution of Carlos - maybe only out of personal interest.
But who of you would like to continue and extend this brilliant solution?

If so - please leave a short message here! There is no duty with it. Just tell about you and your desires, what would you want to be changed?
Also requests of fans of RP are welcome who personally do not want or are able to contribute!

I would be glad to read from you - greetz from Switzerland - Chris

**Comment by Christoph21x** [comment]
I'd like to push this project further ahead - with a strong link to the creator Carlos.

My configuration: I'm not using WMC, using Fortherecord with XBMC front-end.
Reason: I'm still using RP to stream all recordings (*.ts) which can be streamed under "Videos". EPG and recordings are - of course - not accessible.

Wishes: 
1. Expand Remote Potato to be used as streaming server for Fortherecord. Enable EPG and recording's planning
2. Expand Remote Potato to be used as a server for XMBC (including a XBMC client plug-in)

Chris

**Comment by ianken** [comment]
My main objective in forking the bits was to fix the frame rate bug (it was hard coded to 25Hz, which is not cool for those of us in North America). That is fix is working and deployed on my MC box. 

I've also got a private build where I override the settings sent up from the client. I key off the requested bit-rate to select an optimal (to me) configuration for encoding. My devices are all fairly capable so I have enabled features like CABAC and b-frames. The boost in streaming quality both in terms of smoothness and fidelity is substantial over the release bits.

When I find the time (work and family make me time constrained) I intended to add a "Server Side Override" tab to the server settings app to allow the overrides to be configured.

Additionally, I want to add the missing options to the XML blob schema the server and clients pass around to allow RMC to select things like CABAC based on actual phone capabilities. I've not actually investigated that part of the server, so I'm not sure what that would take.

I want to dig more into the windows media streaming engine for non HLS clients (like Windows Phone). I;ve looked at it some and it uses baked in profiles that it loads from a resource. The client has very little control. WP8 devices will be very capable, so the current defined profiles are sub optimal for a modern device. 

Finally finally: I'm not a linux guy. It would be nice if someone could rebuild the ffmpeg variant and bring it up to date with the current x.264 bits. IE: WTV support is now baked in and does not need to be added via patches.

My primary client is Remote Media Center on Android. 

I've alos noticed that the video streamed over the web page is pretty jerky compared to any of my devices. I suspect it's the Silverlive bits. If Carlos has a more up to date version that would be great. If people don't care about that interface it might be worth it to just remove it.

My $.02.

**Comment by garethrepton** [comment]
I'm the developer of RemoteTV on Android. I'm not sure if I'll have much time to contribute to the server but, here are some of my wishes for the project:
1. It would be really, really good to give remote potato a plugin system with a well documented API for the recording scheduling server side of things. Doing this would make creating a 4TheRecord scheduling system much cleaner and give other developers the chance to write plugins (e.g. MediaPortal?).
2. Improve the overall stability of the server, especially some of the API calls (also fix known bugs such as the login hashing).
3. Maybe create a WebService API (WCF?) rather than the current HTTP parsing that's used. It would provide a better base for clients to start with.
4. Improve the streaming support (possibly by updating ffmpeg), currently there are recordings that it won't stream.

**Comment by Christoph21x** [comment]
Hi guys, thanks for your contribution!
Any proposal of how to start?
Did anyone of you changed something so far?

@ianken: I use Silverlight, but only the windowed display brings up the full performance, somehow the full screen version is worse

**Comment by ianken** [comment]
Other than the frame rate fix I have not checked in any other changes. I have my personal build, but it's a pile of hard coded tweaks that'd you'd not want to check in.

-----Original Message-----

**Comment by sccrgoalie1** [comment]
I would like to help.  I've developed the media center addin for remote potato.  I'd really like to help specifically with the recording side of things.  Series management, fix recording request issues, etc.  I've got most of the recording request issues fixed in my custom build.

**Comment by ianken** [comment]
FWIW: things get wonky on Windows 8. Naturally. I can build, run and connect to the server but any attempt to stream fails. As far as I can tell streaming is not working but probing and spawning FFMPEG is. I’ve spent a few minutes looking at it and it’s not obvious yet what’s going on. VS also complains about the bitness of bdatunepia and  mcstoredb in the setup project, does not appear to be an issue.

However, I was able to debug in and saw that the probe calls were working. So it is able to spawn ffmpeg to pull file properties from the WTV file. 

TL/DR: win8 is a hassle.

Edit: Hrrrmmm. It looks like I might be having build environment problems. Got them sorted out but lost some work. Frak.

Edit 2: Got it running on Win8 and Win7. For now it needs two different builds. I've not tested extensively but HLS streaming is fine. Not sure if I had a build problem, or just stupid mistakes. 

-Ian

**Comment by ianken** [comment]
One more note: I'd be fine with yanking out the web interface and the WM streaming bits converging on a well tuned HLS server with support for Media Center.

If any of you are Linux guys, a refreshed version of the FFMPEG bits to bring things up to date with the x264 project would be a good thing, IMHO.

**Comment by sccrgoalie1** [comment]
Glad you got it working.  I hope Carlos gets back to me soon so I can commit some of my changes.  I've got premieres working and am working to perfect series management.  The current approach I took for retrieving series information is to add the necessary information to the RPRequest object returned as part of the RPRecordingBlob.  I've confirmed none of the changes affect any legacy app processing.  I think I'll be creating a new object to send a series request update.  I have silverlight of course, android app (remote media center), iPhone app, and iPad app.  I would like to see these changes get added to the android app (remote media center).

**Comment by garethrepton** [comment]
Good to hear windows 8 support is in the works and possibly working. 

When you say perfecting series management, what enhancements have you made? Just wondering if it's something I might be able to include in my android client when I get a chance.

Also does anyone know if updating ffmpeg in the server might fix some streaming on some recordings? Not sure if this would require a new build of ffmpeg or not.

**Comment by piet5211** [comment]
sccrgoalie1, I will be glad to add the changes to Remote Media Center. Please keep me informed.

**Comment by spokanedj** [comment]
I'm the original author of Smart Potato and worked with Carlos on trying to get his custom HLS Silverlight code running on WP7, with Microsoft's assistance.  Unfortunately because of a job change and other constraints, I wasn't able to resolve the issues, even with MS's help.   I have talked with Carlos extensively about the API and would be happy to take over the responsibility of managing (and possibly creating a new Web API/MVC4 interface) the client API side of things.

My recommendation would be to have a couple people responsible for certain parts of this project, as its probably too big of a project for a single person or a couple people to manage.  In my current job I work extensively writing backend services in .Net for Client and web applications, so I am very comfortable with that section.  Plus with my experience writing Smart Potato I have insight on some of the updates that could make the clients more powerful.

I'm going to take a peak at ianken's fork and would be happy to look at any other modifications that people have made.

**Comment by jchappo** [comment]
I have been working on and off on a RP client for Windows Phone for awhile.  After I complete the initial release, I plan on working on the server a bit.  I have looked through the low level code and plan on adding support for series management, additional EPG content, and possibly a WCF interface to streamline web access.

**Comment by piet5211** [comment]
Hi, I am trying to push my changes to the RP server to my local fork of github, but due to my inexperience with GIT (and Github) I made some stupid mistakes, such as a rebase. I just hope I haven't done anything wrong and/or irreversible.
In the meantime I will temporarily upload my code as a separate project. (RemotePotatoLiveTV)

**Comment by spokanedj** [comment]
You should just need to do a commit (typically to master), then a push to send the changes up to github.  If you haven't got the Windows client (windows.github.com) I would recommend it.  Once you clone your repo, it will monitor it and allow you to commit the changes.  Once you have committed, just click the sync at the top to sync with GitHub.   

One other thing to note:  You might want to add a .gitignore to ignore a lot of files.  The Github client will not let you sync with unchanged files, and just loading the project in VS will create a bunch of them.  I went to tools, settings and created a default .gitignore file to ignore most of them.   I would be happy to cleanup the repo if someone wants to give me write access to the project.  AFAIK, you can't actually make any changes to the original project unless you are given permissions or submit a pull request.   You should create a fork from the main project and submit your changed to your fork.  This allows the original owner to merge those changes if they want.

**Comment by garethrepton** [comment]
@ianken, did you make any further progress with Windows 8? I couldn't see any check-ins on your fork(I can see above you said you didn't want to initially), I've taken a brief look at it today, and it looks like the client needs to be rebuilt on Windows 8 at the least before it will work.

Anyway, I've created a Windows 8 support issue in the issue, I guess we could shove the initial Windows 8 required work in there.

**Comment by ianken** [comment]
To run on 8 I had to build on 8, but I've not investigated further.

Alas real life has intruded into my hacking/free time of late...

-----Original Message-----

**Comment by garethrepton** [comment]
Just built on 8 and can confirm as you say it builds and runs (as an App). Unfortunately I don't have anything to build the service installer.

**Comment by ianken** [comment]
IIRC I was able to build the installer. I had to resolve some resource references that the were missing in the project. They were some images referenced on paths that were invalid. Fix those and you can build.

-----Original Message-----

**Comment by antiflash** [comment]
Glad to see progress regarding W8 compatibility. Is there anything you can share with people like me without the knowledge and tools to compile the code? If you can share one of those builds I would gladly test it.

**Comment by ianken** [comment]
I've propped some changes to the solution that should resolve any build issues. You should be able to sync and load the solution and it should build and "just work." Haven't toughly tested it.

Note: you still need to build on Win8 for bits that run on 8 and ditto for Win7. Since the project take a dependency on in-box Media Center components and they are not identical between 7 and 8 I really don't see a way around this. Run 7 or 8 in a VM, or use another machine.

Anyway, time is limited for me at the moment.

**Comment by drewfreyling** [comment]
I'm happy to help moderate pull requests, bug fixes and the like?

**Comment by ianken** [comment]
I think the author of "remote media center" for android had been hacking on it as well to improve his live tv feature. On his forums there's been some discussion on updating the ffmpeg build and some bug fix proposals, but other than my fix for the frame rate bug I don't think anything has been done. 

The project has a ton of components and it's pretty easy to break things.
